lib/ret_web/controllers/api/v1/media_search_controller.ex

```diff
@@ -54,4 +64,32 @@ defmodule RetWeb.Api.V1.MediaSearchController do
   def index(conn) do
     conn |> send_resp(422, "")
   end
+
+  defp user_assets_index(conn, account, %{"source" => "assets"} = params) do
+    {:commit, results} =
+      %Ret.MediaSearchQuery{source: "assets", user: account.account_id, type: "all", cursor: params["cursor"] || 1} |> Ret.MediaSearch.search()
+
+    conn |> render("index.json", results: results)
+  end
+
```

Comment:
once you refactor the schema to use enums, I'd DRY this up a bit by just checking which enum value to use for the `type` in a lookup table and extracting the rest into its own function